### ISSUE TITLE
Spacing in a macro

### ISSUE THREAD

**Issue by HowardCohl** [brucemiller/LaTeXML#734]
I was wondering if you could give me some advice:

I have defined the following macro:

``\defSpecFun{CFK}[3]{\mathrel{\substack{#3\\[1.5pt]\text{\huge{K}}\\[1.5pt]#1=#2}}}[meaning=semantic-continued-fraction]{2}[\frac{#1}{#2}]``

For instance, here is some LaTeX

``\CFK{i}{1}{10}@{x_i}{y_i}=\sum_{n=1}^{2} f_n``

It looks great when you use pdflatex.

![pdflatex](http://hcohl.sdf.org/Screenshot-14.png)

But if I render using LaTeXML, the spacing (the K is too small) isn't as good.

![LaTeXML](http://hcohl.sdf.org/Screenshot-15.png)

Do you know any ways to customize the macro to make the spacing look better?

**Comment by dginev** [comment]
Is that math rendered as MathML by Firefox, or is it rendered by MathJax? Looks like firefox.

The renderer usually plays a bigger part with determining spacing than the conversion tool, so it may need extra styling or opening issues for firefox/mathjax.

**Comment by HowardCohl** [comment]
It's rendered as MathML using Firefox.

**Comment by dginev** [comment]
A quick fix is to have MathJax rendering the MathML on the page, a slower fix is to adjust the CSS to include math-friendly fonts and adjust any sizing issues by hand. Could be beneficial for LaTeXML's CSS to have that by default, but the general state of math-on-the-web makes this a contentious problem to solve.

**Comment by physikerwelt** [comment]
@dginev I had looked into that recently with Frederic. It seems that there is not too much one can do from the server side. It seems that especially with linux you have to take the effort of setting up the fonts and configuring your Firefox instance.

**Comment by dginev** [comment]
oh joy.

**Comment by fred-wang** [comment]
That's not really true. You can still configure the font and CSS server-side, although it is not always a good idea to override user's preference.

 Howard clearly has a problem with his font set up. However, spacing issues can also come from the converter. Without the MathML output it's impossible to say what the issue is about.

**Comment by HowardCohl** [comment]
Here is the MathML

```
<math alttext="{\displaystyle\mathop{\mathrel{\substack{10\\
\text{\huge{K}}\\
i=1}}\/}\nolimits\frac{x_{i}}{y_{i}}=\sum_{n=1}^{2}f_{n}}" class="ltx_Math" id="p1.1.m1.1" xmlns="http://www.w3.org/1998/Math/MathML">
  <semantics id="p1.1.m1.1a">
    <mrow xref="p1.1.m1.1.8.cmml" id="p1.1.m1.1.8">
      <mrow xref="p1.1.m1.1.1.1.cmml" id="p1.1.m1.1.1.5">
        <mstyle scriptlevel="+1" xref="p1.1.m1.1.1.2.1.1.cmml" id="p1.1.m1.1.1.2.1.2">
          <mtable xref="p1.1.m1.1.1.2.1.1.cmml" id="p1.1.m1.1.1.2.1.2a" rowspacing="0.2ex" columnspacing="0.4em">
            <mtr xref="p1.1.m1.1.1.2.1.1.cmml" id="p1.1.m1.1.1.2.1.2b">
              <mtd xref="p1.1.m1.1.1.2.1.1.cmml" id="p1.1.m1.1.1.2.1.2c">
                <mn xref="p1.1.m1.1.1.2.1.2.2.cmml" id="p1.1.m1.1.1.2.1.2.2">10</mn>
              </mtd>
            </mtr>
            <mtr xref="p1.1.m1.1.1.2.1.1.cmml" id="p1.1.m1.1.1.2.1.2d">
              <mtd xref="p1.1.m1.1.1.2.1.1.cmml" id="p1.1.m1.1.1.2.1.2e">
                <mtext mathsize="207%" xref="p1.1.m1.1.1.2.1.1.1.cmml" id="p1.1.m1.1.1.2.1.2.3" stretchy="false">K</mtext>
              </mtd>
            </mtr>
            <mtr xref="p1.1.m1.1.1.2.1.1.cmml" id="p1.1.m1.1.1.2.1.2f">
              <mtd xref="p1.1.m1.1.1.2.1.1.cmml" id="p1.1.m1.1.1.2.1.2g">
                <mrow xref="p1.1.m1.1.1.2.1.1.cmml" id="p1.1.m1.1.1.2.1.2.4">
                  <mi xref="p1.1.m1.1.1.2.1.2.4.1.cmml" id="p1.1.m1.1.1.2.1.2.4.1">i</mi>
                  <mo xref="p1.1.m1.1.1.2.1.1.1.cmml" id="p1.1.m1.1.1.2.1.2.4.2">=</mo>
                  <mn xref="p1.1.m1.1.1.2.1.2.4.3.cmml" id="p1.1.m1.1.1.2.1.2.4.3">1</mn>
                </mrow>
              </mtd>
            </mtr>
          </mtable>
        </mstyle>
        <mo xref="p1.1.m1.1.1.1.cmml" id="p1.1.m1.1.1.5a">&ApplyFunction;</mo>
        <mstyle xref="p1.1.m1.1.1.1.cmml" id="p1.1.m1.1.1.4" displaystyle="true">
          <mfrac xref="p1.1.m1.1.1.1.cmml" id="p1.1.m1.1.1.4a">
            <msub xref="p1.1.m1.1.1.1.cmml" id="p1.1.m1.1.1.4.2">
              <mi xref="p1.1.m1.1.1.4.2.1.1.cmml" id="p1.1.m1.1.1.4.2.1.1">x</mi>
              <mi xref="p1.1.m1.1.1.4.2.1.2.1.cmml" id="p1.1.m1.1.1.4.2.1.2.1">i</mi>
            </msub>
            <msub xref="p1.1.m1.1.1.1.cmml" id="p1.1.m1.1.1.4.3">
              <mi xref="p1.1.m1.1.1.4.3.1.1.cmml" id="p1.1.m1.1.1.4.3.1.1">y</mi>
              <mi xref="p1.1.m1.1.1.4.3.1.2.1.cmml" id="p1.1.m1.1.1.4.3.1.2.1">i</mi>
            </msub>
          </mfrac>
        </mstyle>
      </mrow>
      <mo xref="p1.1.m1.1.2.cmml" id="p1.1.m1.1.2">=</mo>
      <mrow xref="p1.1.m1.1.8.1.cmml" id="p1.1.m1.1.8.1">
        <mstyle xref="p1.1.m1.1.8.1.1.cmml" id="p1.1.m1.1.8.1.1" displaystyle="true">
          <munderover xref="p1.1.m1.1.8.1.1.cmml" id="p1.1.m1.1.8.1.1a">

            <mo xref="p1.1.m1.1.3.cmml" id="p1.1.m1.1.3" symmetric="true" movablelimits="false" largeop="true">&sum;</mo>

            <mrow xref="p1.1.m1.1.4.1.cmml" id="p1.1.m1.1.4.1">
              <mi xref="p1.1.m1.1.4.1.1.cmml" id="p1.1.m1.1.4.1.1">n</mi>
              <mo xref="p1.1.m1.1.4.1.2.cmml" id="p1.1.m1.1.4.1.2">=</mo>
              <mn xref="p1.1.m1.1.4.1.3.cmml" id="p1.1.m1.1.4.1.3">1</mn>
            </mrow>
            <mn xref="p1.1.m1.1.5.1.cmml" id="p1.1.m1.1.5.1">2</mn>
          </munderover>
        </mstyle>
        <msub xref="p1.1.m1.1.8.1.2.cmml" id="p1.1.m1.1.8.1.2">
          <mi xref="p1.1.m1.1.6.cmml" id="p1.1.m1.1.6">f</mi>
          <mi xref="p1.1.m1.1.7.1.cmml" id="p1.1.m1.1.7.1">n</mi>
        </msub>
      </mrow>
    </mrow>
    <annotation-xml id="p1.1.m1.1b" encoding="MathML-Content">
      <apply xref="p1.1.m1.1.8" id="p1.1.m1.1.8.cmml">
        <eq xref="p1.1.m1.1.2" id="p1.1.m1.1.2.cmml"/>
        <apply xref="p1.1.m1.1.1.5" id="p1.1.m1.1.1.1.cmml">
          <apply xref="p1.1.m1.1.1.2.1.2" id="p1.1.m1.1.1.2.1.1.cmml">
            <csymbol xref="p1.1.m1.1.1.2.1.2.3" id="p1.1.m1.1.1.2.1.1.1.cmml" cd="dlmf">semantic-continued-fraction</csymbol>
            <ci xref="p1.1.m1.1.1.2.1.2.4.1" id="p1.1.m1.1.1.2.1.2.4.1.cmml">i</ci>
            <cn xref="p1.1.m1.1.1.2.1.2.4.3" id="p1.1.m1.1.1.2.1.2.4.3.cmml" type="integer">1</cn>
            <cn xref="p1.1.m1.1.1.2.1.2.2" id="p1.1.m1.1.1.2.1.2.2.cmml" type="integer">10</cn>
          </apply>
          <apply xref="p1.1.m1.1.1.5" id="p1.1.m1.1.1.4.2.cmml">
            <csymbol id="p1.1.m1.1.1.4.2.1.3.cmml" cd="ambiguous">subscript</csymbol>
            <ci xref="p1.1.m1.1.1.4.2.1.1" id="p1.1.m1.1.1.4.2.1.1.cmml">x</ci>
            <ci xref="p1.1.m1.1.1.4.2.1.2.1" id="p1.1.m1.1.1.4.2.1.2.1.cmml">i</ci>
          </apply>
          <apply xref="p1.1.m1.1.1.5" id="p1.1.m1.1.1.4.3.cmml">
            <csymbol id="p1.1.m1.1.1.4.3.1.3.cmml" cd="ambiguous">subscript</csymbol>
            <ci xref="p1.1.m1.1.1.4.3.1.1" id="p1.1.m1.1.1.4.3.1.1.cmml">y</ci>
            <ci xref="p1.1.m1.1.1.4.3.1.2.1" id="p1.1.m1.1.1.4.3.1.2.1.cmml">i</ci>
          </apply>
        </apply>
        <apply xref="p1.1.m1.1.8.1" id="p1.1.m1.1.8.1.cmml">
          <apply xref="p1.1.m1.1.8.1.1" id="p1.1.m1.1.8.1.1.cmml">
            <csymbol id="p1.1.m1.1.8.1.1.1.cmml" cd="ambiguous">superscript</csymbol>
            <apply id="p1.1.m1.1.8.1.1.2.cmml">
              <csymbol id="p1.1.m1.1.8.1.1.2.1.cmml" cd="ambiguous">subscript</csymbol>
              <sum xref="p1.1.m1.1.3" id="p1.1.m1.1.3.cmml"/>
              <apply xref="p1.1.m1.1.4.1" id="p1.1.m1.1.4.1.cmml">
                <eq xref="p1.1.m1.1.4.1.2" id="p1.1.m1.1.4.1.2.cmml"/>
                <ci xref="p1.1.m1.1.4.1.1" id="p1.1.m1.1.4.1.1.cmml">n</ci>
                <cn xref="p1.1.m1.1.4.1.3" id="p1.1.m1.1.4.1.3.cmml" type="integer">1</cn>
              </apply>
            </apply>
            <cn xref="p1.1.m1.1.5.1" id="p1.1.m1.1.5.1.cmml" type="integer">2</cn>
          </apply>
          <apply xref="p1.1.m1.1.8.1.2" id="p1.1.m1.1.8.1.2.cmml">
            <csymbol id="p1.1.m1.1.8.1.2.1.cmml" cd="ambiguous">subscript</csymbol>
            <ci xref="p1.1.m1.1.6" id="p1.1.m1.1.6.cmml">f</ci>
            <ci xref="p1.1.m1.1.7.1" id="p1.1.m1.1.7.1.cmml">n</ci>
          </apply>
        </apply>
      </apply>
    </annotation-xml>
    <annotation id="p1.1.m1.1c" encoding="application/x-tex">{\displaystyle\mathop{\mathrel{\substack{10\\ \text{\huge{K}}\\ i=1}}\/}\nolimits\frac{x_{i}}{y_{i}}=\sum_{n=1}^{2}f_{n}}</annotation>
  </semantics>
</math>
```

**Comment by fred-wang** [comment]
So yes, with the scriptlevel, mathsize and mtable it's hard to tell what will happen... This simpler markup works for me:

        <munderover>
          <mtext mathsize="200%">K</mtext>
          <mrow>
              <mi>i</mi>
              <mo>=</mo>
              <mn>1</mn>
            </mrow>
            <mn>10</mn>
          </munderover>

**Comment by HowardCohl** [comment]
Do you know if there exists a LaTeX equivalent of the above MathML?

**Comment by brucemiller** [comment]
Given the TeX markup you've used, it seems to me that LaTeXML is producing what it should; the spacing problems are due to firefox and/or font issues.  I've noticed that STIX in firefox is particularly bad in this respect.

However, that markup is rather, um, well, ... !?!?  Your macro might look right in the one context you tried it, but you didn't try text mode (or scripts), nor scripts of different heights (it won't center vertically).  Incidentally, Fred's simple solution isn't positioned correctly either.

I assume you've seen ```\sum```? That ought to be the model. Unfortunately, a lot of what makes it work is Knuth's magic fonts and that it's a bigop atom.  You can simulate that with the following:
```
\usepackage{amsopn}
\DeclareMathOperator*{\kettenbruch}{
  \mathchoice%
    {\raisebox{-4pt}{\huge$\operatorname{K}$}}
    {\raisebox{-2pt}{\Large$\operatorname{K}$}}
    {\raisebox{-1.5pt}{\normalsize$\operatorname{K}$}}
    {\raisebox{-1pt}{\scriptsize$\operatorname{K}$}}}
```
Now, just use ```\kettenbruch``` exactly as you would use ```\sum``` or ```\prod```.
This takes care of giving you limits in display mode but scripts otherwise; you get appropriate
font sizes for display, text, script and scriptscript modes; and you get the operator centered on the math axis, as well, thus the raisebox.  

So, the next thing you'd want is to preserve a bit of semantics.  Putting the following in a LaTeXML binding at least makes the symbol itself have an appropriate meaning:
```
DefMath('\kettenbruch',
  '\mathchoice%
    {\raisebox{-4pt}{\huge$\operatorname{K}$}}
    {\raisebox{-2pt}{\Large$\operatorname{K}$}}
    {\raisebox{-1.5pt}{\normalsize$\operatorname{K}$}}
    {\raisebox{-1pt}{\scriptsize$\operatorname{K}$}}',
    scriptpos => \&doScriptpos,
    meaning   => 'continued-fraction',
    mathstyle => \&doVariablesizeOp);
```
Alas, LaTeXML doesn't yet fully understand the semantics of summation or integration limits and bound variables; see #322, but someday it might.

**Comment by HowardCohl** [comment]
This is really good Bruce.  Thanks for the input. I will try this and report back when I have two hands. :)
